### PR TITLE
perfomance/74-revamp Adding parquet for temporary and final files.

### PR DESCRIPTION
**Pull request by lboel**
To Do:
- Ensure correct encoding of columns if required
- Change Documentation information of changed functions
- re-add col_types list if neccessary as forced encoding after parquet read-in
- change final format for all bq files to parquet - maybe also export csv for local testing. 
#74

### CODE REVIEW

File: R/create_table_patient_data_changes_only.R [hunk 1]
```diff
@@ -71,7 +71,7 @@ create_table_longitudinal_data <-
             ungroup() %>%
             arrange(id, tracker_year, tracker_month)
 
-        export_data(
+        export_data_parquet(
```
Comment (pmayd) [inline R/create_table_patient_data_changes_only.R:74]: just a minor thing but I like names that read properly, so I would like to read `export_data_to|as_parquet`

File: R/helper_main.R [hunk 2]
```diff
@@ -157,6 +157,39 @@ export_data <- function(data, filename, output_root, suffix) {
...
+#' export_data(
+#'     data = df_raw_product,
+#'     filename = tracker_name,
+#'     output_root = output_root,
+#'     suffix = "_product_data"
+#' )
+#' }
+export_data_parquet <- function(data, filename, output_root, suffix) {
+    logDebug("Start export_data. Suffix = ", suffix, ".")
+    data %>% arrow::write_parquet(
```
Comment (pmayd) [inline R/helper_main.R:178]: styler should format this differently? I would like to have the pipe always at the end and each function on a new line, so `arrow::write_parquet` should be on its own line and all parameters as well as they are named parameters

File: R/link_product_patient.R [hunk 3]
```diff
@@ -16,8 +16,8 @@
 link_product_patient <- function(product_file, patient_file) {
     logInfo("Trying to link product csv file ", product_file, " with patient csv file ", patient_file)
 
-    patient_data <- read.csv(patient_file, fileEncoding = "UTF-16LE")
-    product_data <- read.csv(product_file, fileEncoding = "UTF-16LE")
+    patient_data <- arrow::read_parquet(patient_file)
+    product_data <-  arrow::read_parquet(product_file)
```
Comment (pmayd) [inline R/link_product_patient.R:20]: space in front of the function name

File: R/read_cleaned_patient_data.R [hunk 4]
```diff
@@ -8,65 +8,11 @@
...
-            "tracker_month" = "i",
-            "tracker_year" = "i",
-            "updated_2022_date" = "D",
-            "weight" = "n"
-        )
-
         patient_data_list <- list()
 
         # get the latest static patient data for each tracker file
         for (patient_file in patient_data_files) {
```
Comment (pmayd) [inline R/read_cleaned_patient_data.R:14]: I guess we could get rid of the for loop here like oleg did in product data and simply use lapply?

File: scripts/run_script_2_clean_data.R [hunk 5]
```diff
@@ -7,8 +7,8 @@ ERROR_VAL_DATE <<- "9999-09-09"
 main <- function() {
     paths <- init_paths(c("patient_data_cleaned", "product_data_cleaned"), delete = TRUE)
     setup_logger(paths$output_root, "script2")
-    patient_data_files <- get_files(paths$tracker_root, pattern = "patient_raw.csv$")
-    product_data_files <- get_files(paths$tracker_root, pattern = "product_raw.csv$")
+    patient_data_files <- get_files(paths$tracker_root, pattern = "patient_raw.parquet$")
```
Comment (pmayd) [inline scripts/run_script_2_clean_data.R:10]: I guess I made an error here as pattern is a regex and "." should be "\\." as it has to be escaped like I did in the other file below, but yeah...works in this case anyway

**Review by pmayd — APPROVED** [review]
LGTM

File: R/helper_main.R (hunk 2 above)
Comment (lboel) [reply R/helper_main.R]: resolved (some linebreak resulted in styler not working properly)

File: R/read_cleaned_patient_data.R (hunk 4 above)
Comment (lboel) [reply R/read_cleaned_patient_data.R]: used purrr (because we have it anyway)

**Comment by lboel** [comment]
ready for merging, if you @pmayd  change the bq scripts to work with parquet

**Pull request merged**